pkg/database/postgresql.go

```diff
@@ -27,11 +27,31 @@ func (info PostgreSQLDatabaseInfo) GetConnectionString() string {
 }
 
 func (info PostgreSQLDatabaseInfo) GetResourcesSQL() string {
-	return "SELECT data FROM resource WHERE kind=$1 AND api_version=$2"
+	return "SELECT data->'metadata'->'creationTimestamp', uuid, data FROM resource WHERE kind=$1 AND api_version=$2 ORDER BY ((data->'metadata'->'creationTimestamp')::text::timestamptz, uuid)"
```

Comment:
Are we going to need an index on `creationTimestamp`?  Have you checked to see if this query is efficient?

Reply:
I wonder if it makes more sense to create a `BIGSERIAL` PK and use that instead of `creationTimestamp`?

Reply:
What good is it to have `uuid` in the `ORDER BY` clause?  I guess you have to in the case the `creationTimestamp` is equal?  If we use a `BIGSERIAL` for this, I think it would be simpler.

Reply:
> I wonder if it makes more sense to create a BIGSERIAL PK and use that instead of creationTimestamp?

I don't think it makes more sense. If we want to go with the same approach that Kubernetes API has, the order of the items is based on the time they were created on the cluster.
We can't ensure that the order of creation in KubeArchive database is the same as the order of creation of the resource in the cluster... so for me it makes more sense doing it this way.

> What good is it to have uuid in the ORDER BY clause? I guess you have to in the case the creationTimestamp is equal? If we use a BIGSERIAL for this, I think it would be simpler.

Exactly, it is for managing the case that creationTimestamp will be the same. We already had a discussion around this. Check https://github.com/kubearchive/kubearchive/pull/484#discussion_r1785888691

A BIGSERIAL will solve this, but still, I think we should stick with the order of the elements based on creation timestamp to follow coherence with what Kubernetes does

Reply:
Understood.  I didn't think about the implications of ordering based on the cluster creation timestamp.

And I will admit I didn't read all the comments.  There are too many, which tells me there probably should have been a design document for this.

Reply:
I created a design document this morning, it adds a little bit of context, not sure if it helps. https://docs.google.com/document/d/1F21wq8JdPGfgQfWH7YrCOZwF9G4u6OgnRs2XndBCero/edit